src/v/crypto/include/crypto/ossl_context_service.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "ssx/thread_worker.h"
+
+#include <memory>
+
+namespace crypto {
+using is_fips_mode = ss::bool_class<struct is_fips_mode_tag>;
+/**
+ * Service used to configure shard local OpenSSL library contexts.
+ * These contexts will be fetched by the OpenSSL library when performing
+ * cryptographic operations.
+ */
+class ossl_context_service final {
```

Comment:
What is the lifecycle of this service? Is it bound to the full application lifecycle? I'm assuming it exists on every core? Should we mention the special responsibilities of core0?

Reply:
I'll add that to the comment, but yes full application lifecycle